blog/_posts/2025-04-22-Enhance-security-OS-Security-Patch.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Enhancing Your Operating System's Security with OS Security Patches in AKS"
+description: "Learn the what, the whys, and the hows of configuring OS Security Patch Auto upgrade channel.  The article also covers some reasons for OS Security Patch, with some primary benefits being increased performance, better security, and minimal disruption to workloads."
+date: 2025-04-22
+authors:
+- Kaarthikeyan Subramanian
+- Ye Wang
+categories: 
+- Security
+- Operations
+---
+
+
+## Why Do We Need Security Patches at the Node Operating System Level?
+
+Operating System (OS) security patches are crucial for maintaining the integrity and security of your systems. These patches address vulnerabilities that could be exploited by malicious actors, ensuring that your system remains protected against the latest threats. Regular OS security patches safeguard data, ensure compliance, and prevent disruptions. At Azure Kubernetes service we are constantly working hard and innovating to guarantee maximum security for your Node operating systems. 
+Presently there are two ways to apply security patches at the OS level automatically with AKS. Read more [OS Security Patch](https://learn.microsoft.com/azure/aks/auto-upgrade-node-os-image?tabs=azure-cli)
+
+**Auto Node Image Channel** - AKS updates nodes weekly with a newly patched VHD for security and bug fixes. This update, which follows maintenance windows and surge settings, incurs no extra VHD cost. Linux unattended upgrades are disabled by default when using this channel. Node image upgrades are supported as long as the cluster's Kubernetes minor version is in support. These AKS-tested node images are fully managed and applied with safe deployment practices.
+**OS Security Patch Channel** - Several customers may need only the security packages for their OS without additional bug fixes and updates. Previously, customers relied on nightly updates and reboot mechanisms, such as KURED, for security updates. Untested package outages highlighted the need for a managed security patch-only channel at the node level. The OS Security Patch channel provides a fully managed, attended Node OS security-only solution. It automates manual processes, eliminating the need for tools like KURED. The Security-Patch channel reimages nodes only when necessary and provides live patch updates with zero disruption, respecting planned maintenance windows.
```

Comment:
the acronym is Kured or KuReD IIRC ?

Reply:
Similarly to above you're telling things that are not necessary now to an audience that at large doesn't even know these things. I don't disagree with mentioning, but I suggest a longer and more comprehensive preamble that describe how patching is done in traditional VMs, and in K8s nodes, there you can list and expand all these concepts and then here say how you're solving / not requiring them.

Also it's quite significant to explain why "eliminating" (probably better word needed) kured is "good" at all. Otherwise begs the question " why would I want to eliminate kured and unattended upgrades" if they do the job?

Reply:
Finally above you mention VHD cost, whereas here you don't. Additionally given below you dedicate sections on comparison, maybe instead of bringing that info here it simply should be mentioned here, and more on the sections below.

Reply:
No its kured https://github.com/kubereboot/kured

Reply:
Removed vhd costing from here and added it under cost comparison in the table. Added a preamble on how security patching was done traditionally and where the gaps are before delving into managed options at AKS.